src/main/src/main/java/org/geoserver/security/urlchecks/GeoServerURLChecker.java

```diff
@@ -50,12 +52,34 @@ public boolean confirm(String url) {
 
             // if enabled but no checks configured, deny access
             List<AbstractURLCheck> enabledUrlList = getEnabledChecks();
-            if (enabledUrlList.isEmpty()) return false;
-
-            for (AbstractURLCheck u : enabledUrlList) {
-                if (u.confirm(url)) {
-                    LOGGER.log(Level.FINE, () -> "URL " + url + " was matched by " + u);
-                    return true;
+            if (!enabledUrlList.isEmpty()) {
```

Comment:
Oh no, a long nested block to avoid an early exit statement? I'm horrified.... :rofl:  (seriously, nested blocks should not be big).